### PR TITLE
dnsdist: Prevent dnsheader alignment issues

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Follow up on the possible `dnsheader` alignment issue mentioned by @omoerbeek in #13280.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdist-lua-actions.cc [hunk 1]
```diff
@@ -242,37 +243,48 @@ std::map<std::string,double> TeeAction::getStats() const
 void TeeAction::worker()
 {
   setThreadName("dnsdist/TeeWork");
-  char packet[1500];
-  int res=0;
-  struct dnsheader* dh=(struct dnsheader*)packet;
-  for(;;) {
-    res=waitForData(d_fd, 0, 250000);
-    if(d_pleaseQuit)
+  std::array<char, 1500> packet;
```
Comment (Habbie) [inline pdns/dnsdist-lua-actions.cc:246]: I know this was already there, but I wonder if 1500 should be a constant somewhere

(I see how this code is safe, to be clear, with `sizeof()` before and `.size()` now)

File: pdns/dnsdist-lua-actions.cc [hunk 2]
```diff
@@ -902,7 +926,10 @@ DNSAction::Action SpoofAction::operator()(DNSQuestion* dq, std::string* ruleresu
     memcpy(dest, recordstart, sizeof(recordstart));
     dest += sizeof(recordstart);
     memcpy(dest, wireData.c_str(), wireData.length());
-    dq->getHeader()->ancount++;
+    dnsdist::PacketMangling::editDNSHeaderFromPacket(dq->getMutableData(), [](dnsheader& header) {
+      header.ancount++;
```
Comment (Habbie) [inline pdns/dnsdist-lua-actions.cc:930]: should this take network/host order into account?

File: pdns/dnsdist-lua-actions.cc [hunk 3]
```diff
@@ -935,11 +965,18 @@ DNSAction::Action SpoofAction::operator()(DNSQuestion* dq, std::string* ruleresu
...
+        header.ancount++;
+        return true;
+      });
     }
   }
 
-  dq->getHeader()->ancount = htons(dq->getHeader()->ancount);
+  auto finalANCount = dq->getHeader()->ancount;
+  dnsdist::PacketMangling::editDNSHeaderFromPacket(dq->getMutableData(), [finalANCount](dnsheader& header) {
+    header.ancount = htons(finalANCount);
```
Comment (Habbie) [inline pdns/dnsdist-lua-actions.cc:977]: ah, bytes are flipped back here :)

**Review by Habbie — APPROVED** [review]
nice, good cleanup too. I left a few comments but none of them are really about the work in this PR.

File: pdns/dnsdist-lua-actions.cc (hunk 1 above)
Comment (rgacogne) [reply pdns/dnsdist-lua-actions.cc]: Good one, will do!

File: pdns/dnsdist-lua-actions.cc (hunk 3 above)
Comment (rgacogne) [reply pdns/dnsdist-lua-actions.cc:983]: Yes :)

**Pull request merged**